aspnetcore/blazor/hybrid/tutorials/windows-forms.md

```diff
@@ -0,0 +1,270 @@
+
+:::image type="content" source="windows-forms/_static/configure-project.png" alt-text="Configure the project.":::
+
+In the **Additional information** dialog, select the framework version, which must be .NET 6.0 or later. Select the **Create** button.
+
+In **Solution Explorer**, right-click the project's name, `WinFormsBlazor` and select **Edit Project File** to open the project file (`WinFormsBlazor.csproj`).
+
+Make the following changes to the project file:
+
+* Change the SDK to `Microsoft.NET.Sdk.Razor`:
+
+  ```xml
+  <Project Sdk="Microsoft.NET.Sdk.Razor">
+  ```
+
+* Add the following configuration inside of the closing `</Project>` tag. 
+
+  ```xml
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Components.WebView.WindowsForms" Version="6.0.200-preview.12.2441" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Content Update="wwwroot\**">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
+  </ItemGroup>
+  ```
+
+Add a `Pages` folder to the Windows Forms project.
+
+Add the following imports file to the root of the Windows Forms project.
+
+`_Imports.razor`:
+
+```razor
+@using System.Net.Http
+@using System.Net.Http.Json
+@using Microsoft.AspNetCore.Components.Forms
+@using Microsoft.AspNetCore.Components.Routing
+@using Microsoft.AspNetCore.Components.Web
+@using Microsoft.AspNetCore.Components.Web.Virtualization
+@using Microsoft.JSInterop
+@using WinFormsBlazor
+@using WinFormsBlazor.Data
+@using WinFormsBlazor.Shared
+```
+
+From the `BlazorWebAssemblyDonor` folder project, import the following folders into the root of the Windows Forms project:
+
+* `Shared`
+* `wwwroot`
+
+In the Windows Forms project, delete the `wwwroot/sample-data` folder, which isn't used.
+
+Open the `wwwroot/index.html` file in the Windows Forms project and change the Blazor script to the following:
+
+```html
+<script src="_framework/blazor.webview.js"></script>
+```
+
+From the `BlazorServerDonor` folder project:
+
+* Import the `Data` folder into the root of the Windows Forms project.
+* From the donor project's `Pages` folder, import the following components into the Windows Forms project's `Pages` folder:
+  * `Counter.razor`
+  * `FetchData.razor`
+  * `Index.razor`
+
+Add the following `Main` component to the Windows Forms project's `Pages` folder.
+
+`Pages/Main.razor`:
+
+```razor
+<Router AppAssembly="@typeof(Program).Assembly" PreferExactMatches="@true">
+    <Found Context="routeData">
+        <RouteView RouteData="@routeData" DefaultLayout="@typeof(MainLayout)" />
+    </Found>
+    <NotFound>
+        <LayoutView Layout="@typeof(MainLayout)">
+            <p>Sorry, there's nothing at this address.</p>
+        </LayoutView>
+    </NotFound>
+</Router>
+```
+
+Open the `Form1.Designer.cs` file from **Solution Explorer**:
```

Comment:
I don't think any modifications of Form1.Designer.cs are necessary, so let's also drop this section.